### ISSUE TITLE
More panels support for plugin

### ISSUE THREAD

**Issue by tenmozes** [VictoriaMetrics/victorialogs-datasource#61]
That would be awesome to add support for the following panel types
- stat
- pie 
- time series
- gauge
- bar

**Comment by pgassmann** [comment]
I managed to create a nice dashboard with pie and timeseries visualization.
the data transformation needed is not trivial though.
![grafik](https://github.com/user-attachments/assets/66cd0eff-7185-4766-b3fb-df6321a6e8a8)

Dashboard Definition Code:
[logs-dashboard-victorialogs.json](https://github.com/user-attachments/files/16798551/logs-dashboard-victorialogs.json)

Issues:
- several Transformations needed to get data in the right format for the visualizations:
Pie chart query:
```
host:~'^$host$' and compose_project:~'^$compose_project$' and compose_service:~'^$compose_service$' and $log_query
  | stats by (host) count() logs
```
Transformations:
![grafik](https://github.com/user-attachments/assets/d26cd9c6-2b45-44ed-a6df-59e3cda08aeb)

Time series:
```
host:~'^$host$' and compose_project:~'^$compose_project$' and compose_service:~'^$compose_service$' and $log_query
  | stats by (_time:1h, host) count() logs
```
![grafik](https://github.com/user-attachments/assets/b340085b-8c86-4c9b-8f98-6aa25749cf49)

- `$__interval` could not be used. `| stats by (_time:$__interval, host) count() logs` is forwarded in the query without being evaluated.

Positive: 
- it's possible now to create dashboards with variables
- its possible to visualize log amounts in timeline
- victorialogs works much faster than loki! Loki is not responding in time when querying just a subset of the same ingested logs.

**Comment by pgassmann** [comment]
@hagen1778 can you add support for the `$__interval` variable in queries?

Are there plans to simplify timeseries queries and other numeric queries for victorialogs?

**Comment by hagen1778** [comment]
Stats panel support depends on https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6942
Time series panel support depends on https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6943

**Comment by hagen1778** [comment]
> @hagen1778 can you add support for the $__interval variable in queries?

@Loori-R do we have support for this variable? If no, would you mind creating a separate feature request for it?

**Comment by pgassmann** [comment]
@hagen1778 the $__interval variable support was just added. https://github.com/VictoriaMetrics/victorialogs-datasource/pull/69

**Comment by dmitryk-dk** [comment]
> > @hagen1778 can you add support for the $__interval variable in queries?
> 
> @Loori-R do we have support for this variable? If no, would you mind creating a separate feature request for it?

I have merged the PR and made a follow-up [PR](https://github.com/VictoriaMetrics/victorialogs-datasource/pull/70) to provide CHANGELOG and tests.

**Comment by dmitryk-dk** [comment]
> I thins this function can be reused https://github.com/VictoriaMetrics/victoriametrics-datasource/blob/main/pkg/plugin/step.go#L113
> 
> but loki can use this type variable https://github.com/grafana/grafana/blob/main/pkg/tsdb/loki/parse_query.go#L25

We have similar logic in the victoriametrics-datasource. There were implemented parsing of the different templates

**Comment by valyala** [comment]
FYI, the [`/select/logsql/stats_query_range`](https://docs.victoriametrics.com/victorialogs/querying/#querying-log-range-stats) endpoint at VictoriaLogs is implemented in the commit [0a40064a6fa64f9d3e3ae60f6f12a9584dce459d](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/0a40064a6fa64f9d3e3ae60f6f12a9584dce459d) . See [these docs](https://docs.victoriametrics.com/victorialogs/querying/#querying-log-range-stats) for details. This endpoint must be used by the VictoriaLogs datasource for building time series panel.

Also the [`/select/logsql/stats_query`](https://docs.victoriametrics.com/victorialogs/querying/#querying-log-stats) endpoint must be used by VictoriaLogs datasource for building stat, pie chart and gauge panels.

**Comment by crocchietta** [comment]
> FYI, the [`/select/logsql/stats_query_range`](https://docs.victoriametrics.com/victorialogs/querying/#querying-log-range-stats) endpoint at VictoriaLogs is implemented in the commit [0a40064a6fa64f9d3e3ae60f6f12a9584dce459d](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/0a40064a6fa64f9d3e3ae60f6f12a9584dce459d) . See [these docs](https://docs.victoriametrics.com/victorialogs/querying/#querying-log-range-stats) for details. This endpoint must be used by the VictoriaLogs datasource for building time series panel.
> 
> Also the [`/select/logsql/stats_query`](https://docs.victoriametrics.com/victorialogs/querying/#querying-log-stats) endpoint must be used by VictoriaLogs datasource for building stat, pie chart and gauge panels.

Are there any plans in the short term to take this into account in victorialogs-datasource?

**Comment by Loori-R** [comment]
Hi @crocchietta!

Thank you for your question. This is not planned in the near future, but we intend to address it later. The task is complex and will require time.

**Comment by hagen1778** [comment]
https://github.com/VictoriaMetrics/victorialogs-datasource/pull/106 has been merged, but there are following enhancements required for better usability of this feature:

* gauge, bar gague, stat, bar chart panel should have toggle for instant and range requests
* support $__range variable in queries. That would be useful for building queries like `_time: $__range | stats by(host) count()` to get the stats over selected time range in Grafana time picker. // this is probably related to backend implementation cc @dmitryk-dk 
* support legend template. For example, I'd like to choose which labels from the result should be plotted in the legend. Currently, I can't control legend
<img width="344" alt="image" src="https://github.com/user-attachments/assets/c0c36484-5eb8-491b-b652-46396b488f3c">
* stats-like panels should explicitly warn user about adding [`| stats` pipe](https://docs.victoriametrics.com/victorialogs/logsql/#stats-pipe), because these panels can only work with stats API

@dmitryk-dk @Loori-R could you please break it into tickets which suits you the best and implement?

**Comment by dmitryk-dk** [comment]
Hi @tenmozes ! The feature was implemented in the [v0.8.0](https://github.com/VictoriaMetrics/victorialogs-datasource/releases/tag/v0.8.0) release. Close the ticket as implemented. 

Other improvements were created in the other tickets and will be implemented soon
